src/display/text_layer.js

```diff
@@ -213,19 +531,85 @@ var renderTextLayer = (function renderTextLayerClosure() {
       var styles = this._textContent.styles;
       var textDivs = this._textDivs;
       var viewport = this._viewport;
+      var bounds = [];
+      var enhanceTextSelection = this._enhanceTextSelection;
+      var enhanceOnClick = this._enhanceOnClick;
       for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(textDivs, viewport, textItems[i], styles);
+        appendText(textDivs, viewport, textItems[i], styles, bounds,
+                   enhanceTextSelection);
+      }
+      if (enhanceTextSelection) {
+        expand(bounds, this._viewport);
+        if (enhanceOnClick) {
+          this._bindMouse(textDivs);
+        }
       }
-
       if (!timeout) { // Render right away
-        render(this);
+        render(this, enhanceOnClick);
       } else { // Schedule
         var self = this;
         this._renderTimer = setTimeout(function() {
-          render(self);
+          render(self, enhanceOnClick);
           self._renderTimer = null;
         }, timeout);
       }
+    },
+
+    _bindMouse: function TextLayer_bindMouse(textDivs) {
```

Comment:
To me, it seems *very inefficient* to add event listeners to *every* single `textDiv` like this.
In practice, this also means that the expansion will not happen if text-selection is started *outside* of a `textDiv`.

I'm wondering if wouldn't make more sense to register the events on the just the `textLayerDiv` instead? (This might require that a reference to the `textLayerDiv` is passed in to `renderTextLayer`.)